pkg/db/db.go

```diff
@@ -3081,3 +3118,230 @@ func (h *DBHandler) processAllTeamLocksRow(ctx context.Context, err error, rows
 	}
 	return result, nil
 }
+
+type QueuedDeployment struct {
+	EslVersion EslId
+	Created    time.Time
+	Env        string
+	App        string
+	Version    *int64
+}
+
+func (h *DBHandler) DBSelectAnyQueuedVersion(ctx context.Context, tx *sql.Tx) (*QueuedDeployment, error) {
+	if h == nil {
+		return nil, nil
+	}
+	if tx == nil {
+		return nil, fmt.Errorf("DBSelectAnyQueuedVersion: no transaction provided")
+	}
+	span, _ := tracer.StartSpanFromContext(ctx, "DBSelectAnyQueuedVersion")
+	defer span.Finish()
+
+	insertQuery := h.AdaptQuery(
+		"SELECT eslVersion, created, envName, appName, queuedVersion FROM deployment_attempts ORDER BY eslVersion DESC LIMIT 1;")
+
+	span.SetTag("query", insertQuery)
+	rows, err := tx.QueryContext(
+		ctx,
+		insertQuery)
+	return h.processDeploymentAttemptsRow(ctx, rows, err)
+}
+
+func (h *DBHandler) DBSelectQueuedDeploymentHistory(ctx context.Context, tx *sql.Tx, environmentName, appName string, limit int) ([]QueuedDeployment, error) {
```

Comment:
Please rename this to be consistent with other functions, e.g. all should contain "DeploymentAttempt"